### DISCUSSION TITLE
Request/Reply in One-to-Many reactor configuration?

### DISCUSSION THREAD

**Discussion by stustd** [lf-lang/lingua-franca#1253] [Q&A] [answered]
Question about communication idiom and whether it can be implemented in pure LF:

I have one server reactor (SR) and many identical client reactors (CR) instantiated in a clients vector. Each CR at times sends a request event to the SR which should then reply to the requesting client. As I want a deterministic simulation, how can I achieve this within the LF framework (if possible...)? (Alternatively I'm perhaps forced to use socket communication for the reply channel, but I'm not sure yet...).

**Comment by edwardalee** [comment, marked as answer]
See https://www.lf-lang.org/docs/handbook/multiports-and-banks

If you want to map the server and clients onto different machines, see https://www.lf-lang.org/docs/handbook/distributed-execution
